sdk/messaging/azservicebus/internal/namespace.go

```diff
@@ -371,7 +388,13 @@ func (ns *Namespace) negotiateClaimImpl(ctx context.Context,
 			case <-refreshCtx.Done():
 				return
 			case <-time.After(renewalDuration):
-				_ = refreshClaim() // logging will report the error for now
+				expiresOn, err = refreshClaim() // logging will report the error for now
+
+				if err != nil {
+					expiresOn = time.Now().Add(time.Minute) // check again in a minute
+				}
+
+				renewalDuration = time.Until(expiresOn)
```

Comment:
Does this use the token's expiration time as-is, or do we adjust it to make the window smaller so that we attempt to refresh before the token is actually expired?

Reply:
Oof, yeah, you're right I need to do that.